src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImp;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImp;
+import core.basesyntax.exception.DaoAnnotationNotFound;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(BetDao.class)) {
+                    field.set(instance, Factory.getBetDao());
+                }
+                if (field.getType().equals(UserDao.class)) {
+                    field.set(instance, Factory.getUserDao());
+                }
+            }
+        }
+        return instance;
+    }
+
+    public static void checkDaoAnnotation() throws DaoAnnotationNotFound {
```

Comment:
where are you calling this method currently?

let's remove this static method and try to write all checks elegantly